packages/info/index.ts

```diff
@@ -1,29 +1,77 @@
+import chalk from "chalk";
 import * as envinfo from "envinfo";
 import * as process from "process";
+import { argv } from "./options";
 
-/**
- * Prints debugging information for webpack issue reporting
- */
+import { AVAILABLE_COMMANDS, AVAILABLE_FORMATS, IGNORE_FLAGS } from "./commands";
+import { configReader, fetchConfig, resolveFilePath, getNameFromPath } from "./configParser";
+import { renderTable } from "./renderTable";
 
 interface Information {
-	Binaries: string[];
-	Browsers: string[];
-	System: string[];
-	npmGlobalPackages: string[];
-	npmPackages: string | string[];
+	Binaries?: string[];
+	Browsers?: string[];
+	System?: string[];
+	npmGlobalPackages?: string[];
+	npmPackages?: string | string[];
 }
+interface ArgvI {
+	_?: string[];
+	bin?: boolean;
+	binaries?: boolean;
+}
+
+const CONFIG = {};
+const DEFAULT_DETAILS: Information = {
+	Binaries: ["Node", "Yarn", "npm"],
+	Browsers: ["Chrome", "Firefox", "Safari"],
+	System: ["OS", "CPU", "Memory"],
+	npmGlobalPackages: ["webpack", "webpack-cli"],
+	npmPackages: "*webpack*"
+};
+
+let DETAILS_OBJ = {};
 
-// eslint-disable-next-line
-export function information(): Information {
-	return {
-		Binaries: ["Node", "Yarn", "npm"],
-		Browsers: ["Chrome", "Firefox", "Safari"],
-		System: ["OS", "CPU"],
-		npmGlobalPackages: ["webpack", "webpack-cli"],
-		npmPackages: "*webpack*"
-	};
+export function informationType(type: string): Information {
+	switch (type) {
+		case "system":
+			return { System: ["OS", "CPU", "Memory"] };
+		case "binaries":
+			return { Binaries: ["Node", "Yarn", "npm"] };
+		case "browsers":
+			return { Browsers: ["Chrome", "Firefox", "Safari"] };
+		case "npmg":
+			return { npmGlobalPackages: ["webpack", "webpack-cli"] };
+		case "npm":
+			return { npmPackages: "*webpack*" };
+	}
 }
+export default async function info(CustomArgv: object): Promise<void> {
+	const CUSTOM_AGRUMENTS: boolean = typeof CustomArgv === "object";
+	const args: ArgvI = CUSTOM_AGRUMENTS ? CustomArgv : argv;
```

Comment:
Why don't you use `yargs` to parse the arguments?

Reply:
Pls elaborate.

Reply:
`yargs` should be used to parse the arguments passed to the cli. That's what we have. We should also install the `@types/yargs` if we didn't, so we already have the correct interfaces

Reply:
You define the options using yargs, so could you please use yargs to parse the arguments too? It will keep consistency. Sorry to be annoying but I don't see any reason to not do that

Reply:
I am using yargs to parse the arguments. Though I am not using `yargs-parser`.

Reply:
Why not?